src/matches.ts

```diff
@@ -18,7 +22,19 @@ export default class Matches {
    * @memberOf Matches
    */
   constructor(matchId: string, callback: any) {
-    const uri = `${CONFIG.BASE}/${matchId}`
+    if (matchId.split('/').length >= 3) {
+      this.matchId =
+        parseInt(matchId.split('/')[1], 10) > 100
+          ? matchId.split('/')[1]
+          : matchId.split('/')[2]
+    } else {
+      this.matchId = matchId
+    }
+    this.callback = callback
+  }
```

Comment:
This whole thing seems a bit hacky and not very intuitive to me, can you explain?

Reply:
Yes it could be smother but i dont know how else too do it without dropping support for the old system.